dvc/repo/graph.py

```diff
@@ -5,10 +5,17 @@ def check_acyclic(graph):
     import networkx as nx
     from dvc.exceptions import CyclicGraphError
 
-    cycle = next(nx.simple_cycles(graph), None)
+    try:
+        edges = nx.find_cycle(graph, orientation="original")
+    except nx.NetworkXNoCycle:
+        return
 
-    if cycle:
-        raise CyclicGraphError(cycle)
+    stages = set()
+    for from_node, to_node, _ in edges:
+        stages.add(from_node)
+        stages.add(to_node)
+
+    raise CyclicGraphError(list(stages))
```

Comment:
@shcheklein Great point! I guess best way to think about it is a very long pipeline and then someone creates the last stage that has `-o` to the input of the first stage. I don't think head and tail are useful here though, because the directed edges that `find_cycle` returns might start anywhere, since it walks every node to search for cycles. I would leave this as is for now as this is the same behaviour we've had before.